src/lib/expansion/expansion-panel-header.ts

```diff
@@ -86,17 +88,24 @@ export class MatExpansionPanelHeader implements OnDestroy {
 
   constructor(
     renderer: Renderer2,
+    @Optional() accordion: MatAccordion,
     @Host() public panel: MatExpansionPanel,
     private _element: ElementRef,
     private _focusMonitor: FocusMonitor,
     private _changeDetectorRef: ChangeDetectorRef) {
 
+    let changeSubjects = [panel._inputChanges];
+    if (accordion) {
+      changeSubjects.push(accordion._inputChanges);
+    }
+
     // Since the toggle state depends on an @Input on the panel, we
-    // need to  subscribe and trigger change detection manually.
+    // need to subscribe and trigger change detection manually.
     this._parentChangeSubscription = merge(
       panel.opened,
       panel.closed,
-      panel._inputChanges.pipe(filter(changes => !!(changes.hideToggle || changes.disabled)))
+      merge(...changeSubjects).pipe(
+        filter(changes => !!(changes.hideToggle || changes.disabled || changes.togglePosition)))
```

Comment:
You don't need `!!` in conditionals, only when the value is returned or stored for later

Reply:
It looks like `filter` requires the value returned to be `boolean`, and without the coercion the returned type is `SimpleChanges`